spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -1441,6 +1476,37 @@ private void fixTxOffsetsIfNeeded() {
 			return records;
 		}
 
+		private void beforePoll() {
+			if (this.isBatchListener) {
+				batchInterceptBeforePoll();
+			}
+			else {
+				recordInterceptBeforePoll();
+			}
+		}
+
+		private void batchInterceptBeforePoll() {
+			if (this.commonBatchInterceptor != null) {
+				try {
+					this.commonBatchInterceptor.beforePoll(this.consumer);
+				}
+				catch (Exception e) {
+					this.logger.error(e, "BatchInterceptor threw an exception");
```

Comment:
Do we need a more detailed log message to tell us that it happened for `beforePoll()`. Right now the same message is duplicated in few places for `beforePoll()`, `intercept()` and `finishInvoke()`. It feels like exceptions when invoking interceptor are a corner case, not sure if anyone really pays attention to this log.

Reply:
The stack trace is enough to tell us where it happened, but it wouldn't hurt to mention the method name; interceptors should not throw exceptions, but we need to protect against it.